File: lib/src/main/java/com/sd/lib/cache/handler/impl/IntegerHandler.java
package com.sd.lib.cache.handler.impl;

import com.sd.lib.cache.CacheInfo;
import com.sd.lib.cache.handler.BaseCacheHandler;

/**
 * Integer处理类
 */
public class IntegerHandler extends BaseCacheHandler<Integer> {
    public IntegerHandler(CacheInfo info) {
        super(info);
    }

    @Override
    protected byte[] valueToByte(Integer value) {
        return value.toString().getBytes();
    }

    @Override
    protected Integer byteToValue(byte[] bytes, Class<?> clazz) {
        return Integer.valueOf(new String(bytes));
    }

    @Override
    protected String getKeyPrefix() {
        return "integer_";
    }
}

File: lib/src/main/java/com/sd/lib/cache/handler/impl/IntegerHandler.kt
package com.sd.lib.cache.handler.impl

import com.sd.lib.cache.CacheInfo
import com.sd.lib.cache.handler.BaseCacheHandler

/**
 * Integer处理类
 */
internal class IntegerHandler(info: CacheInfo) : BaseCacheHandler<Int>(info, "integer") {
    override fun encodeToByteImpl(value: Int): ByteArray {
        return value.toString().toByteArray()
    }

    override fun byteToValue(bytes: ByteArray, clazz: Class<*>?): Int {
        return String(bytes).toInt()
    }
}